components/lib/dialog/dialog.d.ts

```diff
@@ -353,6 +353,11 @@ export interface DialogProps {
      * The properties of CSSTransition can be customized, except for "nodeRef" and "in" properties.
      */
     transitionOptions?: CSSTransitionProps | undefined;
+    /**
+     * Specifies the default visibility of the dialog.
+     * @defaultValue false
+     */
+    defaultVisible?: boolean | undefined;
```

Comment:
None of this is necessary just use the visible property if its true the dialog will be displayed even on inital page load.

```js
 /**
     * Specifies the visibility of the dialog.
     * @defaultValue false
     */
    visible?: boolean | undefined;
```

Reply:
You're right, but there's an additional aspect to consider. When it comes to closing the dialog, the user would need to manage the dialog's visibility state and an `onHide` function. By introducing the `defaultVisible` prop, could eliminate the need for the parent component to handle state hooks and `onHide` logic.

Since the Dialog component is already managing its own state, there's no need for the parent component to duplicate this effort just to handle initial visibility. Using `defaultVisible` simplifies the parent component and keeps the state management encapsulated within the Dialog component, making it easier to use.

Reply:
@jpranays i would need to see a use case where this makes any sense to add this property.  The showcase examples all show pretty clear usage and require you to control the state with the `visible` and `onHide` properties.  THis new property just adds confusion.

Reply:
> @jpranays i would need to see a use case where this makes any sense to add this property. The showcase examples all show pretty clear usage and require you to control the state with the `visible` and `onHide` properties. THis new property just adds confusion.

Consider a scenario where a dialog should be displayed immediately upon page load, such as an important announcement or a welcome message. In this case, the parent component doesn't need to control the visibility state after the initial render.
While it's true that visible and onHide properties are essential for controlled components, the defaultVisible prop is intended for scenarios where the dialog's initial state is the only concern. It doesn't override or conflict with the existing visible and onHide properties but rather offers an alternative for specific use cases.

I hope this clarifies the value and use case for the defaultVisible prop. I believe it will enhance the flexibility and usability of the Dialog component without adding unnecessary confusion.

Reply:
@jpranays can you open a ticket first with your use case and details so this can be associated to this PR.

Reply:
Ya sure